installinator/src/write.rs

```diff
@@ -640,6 +645,43 @@ impl ControlPlaneZoneWriteContext<'_> {
         // successor.
         let mut transport = StepHandle::ready(transport);
 
+        // Write out a file to indicate that installinator has updated the
+        // dataset. Do this at the very beginning to ensure that installinator's
+        // presence is recorded even if something goes wrong after this step.
+        transport = engine
+            .new_step(
+                WriteComponent::ControlPlane,
+                ControlPlaneZonesStepId::MupdateOverride,
+                "Writing MUPdate override file",
+                async move |cx| {
+                    let transport = transport.into_value(cx.token()).await;
+                    let mupdate_uuid = MupdateOverrideUuid::new_v4();
```

Comment:
Chatted about this briefly live - I'm worried this will create two different UUIDs for each of the two drives (if we create different update engine registrations for each drive).

Reply:
Yeah this is definitely wrong, thanks for catching it -- will fix and add testing for it.